src/video/wayland/SDL_waylandwindow.c

```diff
@@ -1096,6 +1097,10 @@ handle_surface_enter(void *data, struct wl_surface *surface,
                                   sizeof(SDL_WaylandOutputData*) * (window->num_outputs + 1));
     window->outputs[window->num_outputs++] = driverdata;
 
+    if (!window->fractional_scale) {
+        update_scale_factor(window);
+    }
+
     /* Update the scale factor after the move so that fullscreen outputs are updated. */
     Wayland_move_window(window->sdlwindow, driverdata);
     update_scale_factor(window);
```

Comment:
```suggestion
    if (!window->fractional_scale) {
        update_scale_factor(window);
    }
```

and guarded here as fullscreen window moves have to be handled before updating the scale factor.

Reply:
> I forgot to post that I finally did some more extensive testing. I have one tiny issue with fullscreen to follow up on where the older path and this conflict in one specific chain. I'll sort it soon.

It turns out this is a pre-existing issue, I can reproduce it on master.  I'll file a separate bug so we don't derail this.